### ISSUE TITLE
Parallel routes breaks completely after being redirected from a server action function 

### ISSUE THREAD

**Issue by RahulKathayat** [vercel/next.js#65411]
### Link to the code that reproduces this issue

https://github.com/vercel/nextgram 

### To Reproduce

1. Start the application in dev mode 
2. change the page.tsx file iin (.)photos/[id] folder to the below code 
import { redirect } from 'next/navigation';
import { Modal } from './modal';

export default function PhotoModal({
  params: { id: photoId },
}: {
  params: { id: string };
}) {
  return <Modal>
      <div>{photoId}</div>
      <form action={async ()=>{
        "use server";
        redirect('/');
      }}>
        <button type="submit">Submit</button>        
        </form>    
    </Modal>;
}

3.  Now when i click the submit button im being redirected to the homepage but the other parallel routes are not being displayed when i click them instead it just changes the url to that route and no modal pops up 

### Current vs. Expected behavior

i expected the modal to pop up after being redirected to the '/' route but instead it just changes the url and nothing renders

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Home Single Language
  Available memory (MB): 15694
  Available CPU cores: 12
Binaries:
  Node: 20.11.1
  npm: N/A
  Yarn: N/A
  pnpm: 9.0.6
Relevant Packages:
  next: 14.3.0-canary.40 // Latest available version is detected (14.3.0-canary.40).
  eslint-config-next: N/A
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.4.5
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Parallel & Intercepting Routes

### Which stage(s) are affected? (Select all that apply)

next dev (local), next build (local), next start (local), Vercel (Deployed)

**Comment by Stanislaw09** [comment]
Same here
broken on Next `14.2.1`, `14.2.3`
Works properly on `14.1.3`

**Comment by RahulKathayat** [comment]
Yes but in production im still facing the issue even after i downgraded my nextjs version to 14.1.3

**Comment by sbbu** [comment]
also having this issue on `14.2.1` and `14.2.3`

**Comment by ztanner** [comment]
Hi @RahulKathayat --

I believe the issue will be fixed if you add a `app/@modal/page.tsx` component that returns `null`. The bug here is actually that the modal disappears when you redirect to `/` -- the expected behavior is that it will continue to keep whatever was active in the `modal` slot until it matches to a new page, as described [here](https://nextjs.org/docs/app/building-your-application/routing/parallel-routes#active-state-and-navigation). 

I'll be investigating why the modal is disappearing when the server action redirects to `/`.

**Comment by zaru** [comment]
In my case, I was able to open the modal by creating a `@modal/page.tsx` that returns `null`. Thank you.

https://github.com/vercel/next.js/issues/65936#issuecomment-2119516512

**Comment by alphanetEX** [comment]
> Same here broken on Next `14.2.1`, `14.2.3` Works properly on `14.1.3`

thanks im change the version to 14.1.3 and the redirect works
